src/ripple/app/paths/tests/Path_test.cpp

```diff
@@ -1,212 +1,585 @@
-            Amount(5, "USD", alice), log.stream());
-        log << "ripplePathFind alternatives: " << alternatives;
-
-        expectEquals(alternatives.size(), 0);
+        auto const alternatives = findPath(env.ledger, "alice", "bob",
+            {Account("alice")["USD"]}, Account("bob")["USD"](5));
+        expect(alternatives.size() == 0);
     }
 
     void
     test_direct_path_no_intermediary()
     {
         testcase("direct path no intermediary");
-        std::uint64_t const xrp = std::mega::num;
-
-        auto master = createAccount("masterpassphrase", KeyType::ed25519);
-
-        std::shared_ptr<Ledger const> LCL;
-        Ledger::pointer ledger;
-        std::tie(LCL, ledger) = createGenesisLedger(100000 * xrp, master);
-
-        // Create accounts
-        auto accounts = createAndFundAccounts(master, { "alice", "bob" },
-            KeyType::ed25519, 10000 * xrp, ledger);
-        auto& alice = accounts["alice"];
-        auto& bob = accounts["bob"];
-        expectNotEquals(alice.pk.humanAccountID(), master.pk.humanAccountID());
-
-        // Set credit limit
-        trust(bob, alice, "USD", 700, ledger);
-
-        // Find path from alice to bob
-        auto alternatives = findPath(ledger, alice, bob, { Currency("USD") },
-            Amount(5, "USD", bob), log.stream());
-        log << "ripplePathFind alternatives: " << alternatives;
+        Env env(*this);
+        env.fund(XRP(10000), "alice", "bob");
+        env.trust(Account("alice")["USD"](700), "bob");
+
+        auto const alternatives = findPath(env.ledger, "alice", "bob",
+            {Account("alice")["USD"]}, Account("bob")["USD"](5));
+        Json::Value jv;
+        Json::Reader().parse(R"([{
+		        "paths_canonical" : [],
+		        "paths_computed" : [],
+		        "source_amount" :
+		        {
+			        "currency" : "USD",
+			        "issuer" : "rG1QQv2nh2gr7RCZ1P8YYcBUKCCN633jCn",
+			        "value" : "5"
+		        }
+            }])", jv);
+        expect(jv == alternatives);
+    }
 
-        expectEquals(alternatives.size(), 1);
-        auto alt = alternatives[0u];
-        expectEquals(alt[jss::paths_canonical].size(), 0);
-        expectEquals(alt[jss::paths_computed].size(), 0);
-        auto srcAmount = alt[jss::source_amount];
-        expectEquals(srcAmount[jss::currency], "USD");
-        expectEquals(srcAmount[jss::value], "5");
-        expectEquals(srcAmount[jss::issuer], alice.pk.humanAccountID());
+    void
+    test_payment_auto_path_find()
+    {
+        testcase("payment auto path find");
+        Env env(*this);
+        auto const gw = Account("gateway");
+        auto const USD = gw["USD"];
+        env.fund(XRP(10000), "alice", "bob", gw);
+        env.trust(USD(600), "alice");
+        env.trust(USD(700), "bob");
+        env(pay(gw, "alice", USD(70)));
+        env(pay("alice", "bob", USD(24)));
+        env.require(balance("alice", USD(46)));
+        env.require(balance("bob", USD(24)));
```

Comment:
The original js test validated the balance of the gateway, and I think that would be valuable here, too.
```
env.require(balance(gw, Account("alice")["USD"](-46)));
env.require(balance(gw, Account("bob")["USD"](-24)));
```